libraries/cmake/source/boost/CMakeLists.txt

```
@@ -694,11 +694,19 @@ function(generateBoostContext)
       "${library_root}/src/asm/ontop_x86_64_sysv_macho_gas.S"
     )
   elseif(DEFINED PLATFORM_WINDOWS)
-    set(asm_source_file_list
-      "${library_root}/src/asm/jump_x86_64_ms_pe_masm.asm"
-      "${library_root}/src/asm/make_x86_64_ms_pe_masm.asm"
-      "${library_root}/src/asm/ontop_x86_64_ms_pe_masm.asm"
-    )
+    if(CMAKE_SHARED_LINKER_FLAGS STREQUAL "/machine:X86")
```

CMake keeps track of the size of a void pointer, this is commonly used to generically detect bitness of the compiler used.
```suggestion
    if(CMAKE_SIZEOF_VOID_P EQUAL 4)
```

---

Done.